src/main/java/seedu/bigpp/parser/Parser.java

```diff
@@ -9,32 +9,37 @@
 import seedu.bigpp.command.viewercommand.ViewerDeleteCommand;
 import seedu.bigpp.command.viewercommand.ViewerEditCommand;
 import seedu.bigpp.command.viewercommand.ViewerViewCommand;
+import seedu.bigpp.exceptions.UnrecognizedCommandException;
 import seedu.bigpp.ui.UI;
 
 public class Parser {
 
-    public Command parseCommand(String userInput) {
+    public Command parseCommand(String userInput) throws UnrecognizedCommandException {
 
         // try common commands first
-        if (userInput.equals("back")) {
-            return new BackCommand();
-        }
+        try {
+            if (userInput.equals("back")) {
+                return new BackCommand();
+            }
 
-        if (userInput.equals("bye")) {
-            return new ByeCommand();
-        }
+            if (userInput.equals("bye")) {
+                return new ByeCommand();
+            }
 
-        // once common commands are tried, try menu specific commands
-        switch (UI.getUiState()) {
+            // once common commands are tried, try menu specific commands
+            switch (UI.getUiState()) {
 
-        case PCVIEWER:
-            return parseViewerCommand(userInput);
+            case PCVIEWER:
+                return parseViewerCommand(userInput);
 
-        case PCBUILDER:
-            return parseBuilderCommand(userInput);
+            case PCBUILDER:
+                return parseBuilderCommand(userInput);
 
-        default:
-            return null;
+            default:
+                return null;
+            }
+        } catch (Exception e) {
+            throw new UnrecognizedCommandException();
```

Comment:
This `UnrecognizedCommandException()` will be caught when any command has a wrong input, I think this approach is not good as the only reason why we get exceptions for this part is when our "arguments" do not exist in `parseViewerCommand` and `parseBuilderCommand`.

Reply:
yessir